widgyts/dataset_viewer.py

```diff
@@ -199,9 +200,22 @@ def widget(self):
 
         traitlets.link((dropdown, "value"), (self, "grid_colormap"))
 
+        button = ipywidgets.Button(description="Add Keyframe")
+
+        def on_button_clicked(b):
+            self.position_list = self.position_list + [self.renderer.camera.position]
+
+        button.on_click(on_button_clicked)
+
+        select = ipywidgets.Select(
```

Comment:
So I *think* that using `options=self.position_list` won't work.  What I think you should start out with is an empty list.  And then, inside `on_button_clicked`, update it.  For now let's take this in two steps -- the first is to add a new option to the `select` object every time you click the button.  The second step will be to make that correspond to something meaningful.

So the next phase will be, change `options` to an empty list, then inside `on_button_clicked`, update position list *and* add a new item to `select.options`.  But, just add something like `"Hello"` instead of the position list for the `options`.